phpseclib/Crypt/Hash.php

```diff
@@ -213,79 +150,24 @@ function setHash($hash)
                 break;
             case 'sha512':
                 $this->l = 64;
-        }
-
-        switch ($hash) {
-            case 'md2':
-                $mode = CRYPT_HASH_MODE == self::MODE_HASH && in_array('md2', hash_algos()) ?
-                    self::MODE_HASH : self::MODE_INTERNAL;
-                break;
-            case 'sha384':
-            case 'sha512':
-                $mode = CRYPT_HASH_MODE == self::MODE_MHASH ? self::MODE_INTERNAL : CRYPT_HASH_MODE;
                 break;
             default:
```

Comment:
This block is not covered by tests. E.g. add a single test for whirlpool (no truncation) and one for tiger128,3-96 (truncated)?